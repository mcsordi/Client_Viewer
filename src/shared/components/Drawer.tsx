import React, { useContext } from 'react';
import { themeContext } from '../../contexts/ThemeContext/context.ts';
import { ImgContainer } from './ImgContainer.tsx';
import { NavDrawer } from './NavDrawer.tsx';
import { IoMdHome } from 'react-icons/io';
import { FaStar } from 'react-icons/fa';

type ITheme = {
  theme?: boolean;
};

export const Drawer: React.FC<ITheme> = ({ theme }) => {
  const { light } = useContext(themeContext);
  return (
    <div
      className={`${light || theme ? `bg-white` : `bg-zinc-700`}   w-56 min-h-screen`}
    >
      <ImgContainer path="./1678304807285.jpeg" />
      <NavDrawer
        text="PÃ¡gina Inicial"
        icon={<IoMdHome className="text-xl" />}
      />
      <NavDrawer text="star page" icon={<FaStar className="text-xl" />} />
    </div>
  );
};
