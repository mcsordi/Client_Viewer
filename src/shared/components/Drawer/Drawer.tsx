import React, { useContext } from 'react';
import { themeContext } from '../../../contexts/ThemeContext/context.ts';
import { ImgContainer } from '../ImgContainer/ImgContainer.tsx';
import { NavDrawer } from '../NavDrawer/NavDrawer.tsx';
import { IoMdHome } from 'react-icons/io';
import { FaStar } from 'react-icons/fa';
import { toggle } from '../../../contexts/DrawerToggle/context.ts';

type ITheme = {
  theme?: boolean;
};

export const Drawer: React.FC<ITheme> = ({ theme }) => {
  const { light } = useContext(themeContext);
  const { show } = useContext(toggle);
  return (
    <div
      className={`${show ? `flex` : 'hidden'} md:flex flex-col ${light || theme ? `bg-white` : `bg-zinc-700`} w-56 min-h-screen`}
    >
      <ImgContainer path="./1678304807285.jpeg" />
      <NavDrawer
        text="PÃ¡gina Inicial"
        icon={<IoMdHome className="text-xl" />}
      />
      <NavDrawer text="star page" icon={<FaStar className="text-xl" />} />
    </div>
  );
};
