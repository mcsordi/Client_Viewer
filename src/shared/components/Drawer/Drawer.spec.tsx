import { render } from '@testing-library/react';
import { Drawer } from './Drawer.tsx';
import React from 'react';

describe('<Drawer/>', () => {
  it('should render Drawer component in the dark mode', () => {
    const { container } = render(<Drawer theme={false} />);
    expect(container.firstChild).toHaveClass('bg-zinc-700');
  });
  it('should render Drawer component in the light mode', () => {
    const { container } = render(<Drawer theme={true} />);
    expect(container.firstChild).toHaveClass('bg-white');
  });
  it('should render match a snapshot', () => {
    const { container } = render(<Drawer theme={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
