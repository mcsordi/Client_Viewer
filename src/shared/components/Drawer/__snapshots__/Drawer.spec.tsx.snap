// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer/> should render match a snapshot 1`] = `
<div
  class="hidden md:flex flex-col bg-white w-56 min-h-screen"
>
  <div
    class="flex w-full h-44 border-b-2 items-center justify-center"
  >
    <img
      alt="profile image"
      class="w-24 h-24 rounded-full"
      src="./1678304807285.jpeg"
    />
  </div>
  <div
    class="text-gray-200
         cursor-pointer flex w-full justify-between items-center px-11 p-3.5 border-b-2"
  >
    <svg
      class="text-xl"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M208 448V320h96v128h97.6V256H464L256 64 48 256h62.4v192z"
      />
    </svg>
     
    <p
      class="min-w-24 font-medium"
    >
      PÃ¡gina Inicial
    </p>
  </div>
  <div
    class="text-gray-200
         cursor-pointer flex w-full justify-between items-center px-11 p-3.5 border-b-2"
  >
    <svg
      class="text-xl"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
     
    <p
      class="min-w-24 font-medium"
    >
      star page
    </p>
  </div>
</div>
`;
